File: src/demo_cpp_pkg/src/person_node.cpp
#include <string>
#include <rclcpp/rclcpp.hpp>

class PersonNode : public rclcpp::Node
{
private:
  std::string name_;
  int age_;
public:
    PersonNode(const std::string &node_name,const std::string &name,const int &age) : Node(node_name)
    {
        this->name_= name;
        this->age_=age;

    };

    void eat(const std::string &food)
    {
        RCLCPP_INFO(this->get_logger(), "%s ,今年%d岁,is eating %s", this->name_.c_str(), this->age_, food.c_str());
    }

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto person_node = std::make_shared<PersonNode>("person_node","Alice",30);
    person_node->eat("apple");
    rclcpp::spin(person_node);
    rclcpp::shutdown();
    return 0;
}

File: src/demo_python_pkg/demo_python_pkg/person_node.py
import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self,name:str,age:int)-> None:
        super().__init__("node_name")
        self.name = name
        self.age = age  

    def eat(self,food_name:str):
       self.get_logger().info(f"我叫{self.name},我今年{self.age}岁,我正在吃{food_name}  ")

def main():
    rclpy.init()
    node = PersonNode("小明",18)
    node.eat("苹果")
    rclpy.spin(node)
    rclpy.shutdown()